From f7b325764bc9fd9a9275f9aeab2755a480c67e71 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Fri, 22 Jul 2011 10:07:04 +1000
Subject: [PATCH] Always reset the fd to -1.

The server currently resets the fd to -1 only for hotplugged devices, not
xorg.conf devices. These have an fd of 0 and a close can result in an error
log close.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/evdev.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/evdev.c b/src/evdev.c
index 77b19cf..1c8d45a 100644
--- a/src/evdev.c
+++ b/src/evdev.c
@@ -2285,6 +2285,7 @@ EvdevPreInit(InputDriverPtr drv, InputInfoPtr pInfo, int flags)
     pInfo->device_control = EvdevProc;
     pInfo->read_input = EvdevReadInput;
     pInfo->switch_mode = EvdevSwitchMode;
+    pInfo->fd = -1;
 
     rc = EvdevOpenDevice(pInfo);
     if (rc != Success)
-- 
1.7.6

